From 465ae0986aded52132865bd7d254235bf3a27666 Mon Sep 17 00:00:00 2001
From: Vibhore Vardhan <vibhore@ti.com>
Date: Thu, 15 Feb 2024 12:16:09 -0600
Subject: [PATCH 04/27] arm64: dts: ti: k3-am62x: Fix MCU GPIO wakeup pin

LPM wakeup sources overlay had a pin confict between the MCU GPIO and
MCU MCAN1 wakeup sources.

This fix changes the MCU GPIO to a different pin so that both MCU MCAN1
and MCU GPIO wakeup sources can be enabled at the same time.

Fixes: 4ba1886b886d ("arm64: dts: ti: k3-am62: Create a single LPM wkup sources overlay")
Signed-off-by: Vibhore Vardhan <vibhore@ti.com>
---
 arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso b/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso
index fccdbc64d39c..a251544d7807 100644
--- a/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso
+++ b/arch/arm64/boot/dts/ti/k3-am62x-sk-lpm-wkup-sources.dtso
@@ -38,11 +38,11 @@
 		pinctrl-names = "default";
 		pinctrl-0 = <&wake_mcugpio1_pins_default>;
 		interrupt-parent = <&mcu_gpio0>;
-		interrupts = <16 IRQ_TYPE_EDGE_RISING>;
+		interrupts = <4 IRQ_TYPE_EDGE_RISING>;
 		switch {
 			label = "MCUGPIO";
 			linux,code = <143>;
-			gpios = <&mcu_gpio0 16 GPIO_ACTIVE_LOW>;
+			gpios = <&mcu_gpio0 4 GPIO_ACTIVE_LOW>;
 			wakeup-source;
 		};
 	};
@@ -59,7 +59,7 @@
 &mcu_pmx0 {
 	wake_mcugpio1_pins_default: wake-mcugpio1-pins-default {
 		pinctrl-single,pins = <
-			AM62X_MCU_IOPAD(0x0040, PIN_INPUT, 7) /* (D6) MCU_MCAN1_RX.MCU_GPIO0_16 */
+			AM62X_MCU_IOPAD(0x0010, PIN_INPUT, 7) /* (D8) MCU_SPI0_D1.MCU_GPIO0_4 */
 		>;
 	};
 
-- 
2.39.2

